File: src/main/java/eu/monsbot/turbokarts/vehiclelogic/karts/vehicles/AbstractWheel.java
package eu.monsbot.turbokarts.vehiclelogic.karts.vehicles;

import org.bukkit.Material;
import org.bukkit.inventory.ItemStack;
import org.joml.Vector3f;

public abstract class AbstractWheel {
    // Display

    Material material;
    int      customModelData;
    Vector3f translation;
    Vector3f size;
    Vector3f left_rotation_axis;
    float    left_rotation_value;
    Vector3f right_rotation_axis;
    float    right_rotation_value;

    // Rolling
    Vector3f rotationAxis;
    float radius; //in blocks
    boolean rotateClockwise;

    // Steering
    boolean canSteer;
    Vector3f steeringAxis;
}

File: src/main/kotlin/eu/monsbot/turbokarts/vehiclelogic/karts/vehicles/AbstractWheel.kt
package eu.monsbot.turbokarts.vehiclelogic.karts.vehicles

import org.bukkit.Material
import org.joml.Vector3f

abstract class AbstractWheel {
    // Display
    var material: Material? = null
    var customModelData: Int = 0
    var translation: Vector3f? = null
    var size: Vector3f? = null
    var left_rotation_axis: Vector3f? = null
    var left_rotation_value: Float = 0f
    var right_rotation_axis: Vector3f? = null
    var right_rotation_value: Float = 0f

    // Rolling
    var rotationAxis: Vector3f? = null
    var radius: Float = 0f //in blocks
    var rotateClockwise: Boolean = false

    // Steering
    var canSteer: Boolean = false
    var steeringAxis: Vector3f? = null
}
